cmd/run_test.go

```diff
@@ -81,3 +85,89 @@ func TestRunWithoutConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestNoCleanupFlag(t *testing.T) {
+	afs := afero.NewOsFs()
+	testCases := []struct {
+		name             string
+		content          string
+		execConfig       blocks.TTPExecutionConfig
+		expectedDirExist bool
+		wantError        bool
+	}{
+		{
+			name: "Test No Cleanup Behavior - Directory Creation",
+			content: `
+---
+name: test-cleanup
+steps:
+  - name: step_one
+    inline: mkdir testDir
+    cleanup:
+      inline: rm -rf testDir`,
+			execConfig: blocks.TTPExecutionConfig{
+				NoCleanup: true,
+			},
+			expectedDirExist: true,
+			wantError:        false,
+		},
+		{
+			name: "Test Cleanup Behavior - Directory Deletion",
+			content: `
+---
+name: test-cleanup-2
+steps:
+  - name: step_two
+    inline: mkdir testDir2
+    cleanup:
+      inline: rm -rf testDir2`,
+			execConfig: blocks.TTPExecutionConfig{
+				NoCleanup: false,
+			},
+			expectedDirExist: false,
+			wantError:        false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a temp directory to work within
+			tempDir, err := afero.TempDir(afs, "", "testCleanup")
+			require.NoError(t, err)
+
+			// Update content to work within the temp directory
+			tc.content = strings.ReplaceAll(tc.content, "mkdir ", "mkdir "+tempDir+"/")
+			tc.content = strings.ReplaceAll(tc.content, "rm -rf ", "rm -rf "+tempDir+"/")
+
+			// Render the templated TTP first
+			ttp, err := blocks.RenderTemplatedTTP(tc.content, &tc.execConfig)
+			require.NoError(t, err)
+
+			// Handle potential error from RemoveAll within a deferred function
+			defer func() {
+				err := afs.RemoveAll(tempDir) // cleanup temp directory
+				if err != nil {
+					t.Errorf("failed to remove temp directory: %v", err)
+				}
+			}()
+
+			_, err = ttp.RunSteps(tc.execConfig)
+			if tc.wantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			// Determine which directory to check based on the test case content
+			dirName := tempDir + "/testDir"
+			if strings.Contains(tc.content, "testDir2") {
```

Comment:
this will break if you add more than two test cases to the array above, generally would recommend  explicitly specifying the directory to check as part of the test config, as "this test can only ever have two cases" something another developer would expect